libraries/alicloud_backend.rb

```diff
@@ -33,7 +33,12 @@ def alicloud_client(api:, api_version:)
     endpoint ||= if %w{sts ram resourcemanager ims rds}.include?(api)
                    "https://#{api}.aliyuncs.com"
                  else
-                   "https://#{api}.#{region}.aliyuncs.com"
+                   # AliCloud VPN endpoints vary between regions, so the following accounts for that variability
+                   if api == "vpn" && %w{cn-qingdao cn-beijing cn-beijing cn-shanghai cn-shenzhen cn-hongkong ap-southeast-1 us-west-1 us-east-1 cn-shanghai-finance-1 cn-shenzhen-finance-1 cn-north-2-gov-1}.include(region)
+                     "https://#{api}.aliyuncs.com"
+                   else
+                     "https://#{api}.#{region}.aliyuncs.com"
+                   end
```

Comment:
better way to do this is as below, 

```
def region
    region = @client_args && @client_args.fetch(:region, nil)
    region || ENV["ALICLOUD_REGION"]
end

def alicloud_vpn_client
    api = 'vpn'
    api = "#{api}.#{region}" unless %w{cn-qingdao cn-beijing cn-beijing cn-shanghai cn-shenzhen cn-hongkong ap-southeast-1 us-west-1 us-east-1 cn-shanghai-finance-1 cn-shenzhen-finance-1 cn-north-2-gov-1}.include(region)
    alicloud_client(api: api, api_version: "2017-12-04")
end
```

Reply:
this way we can manage new type of clients as well ..

Reply:
This is adding an entirely new method to the resource pack, there is currently no alicloud_vpn_client method - this would require existing uses of the resource pack to be refactored.

Reply:
okay.. considering above concern, will approve this ..